File: UKParliament.CodeTest.Data/ViewModels/EmployeeViewModel.cs
namespace UKParliament.CodeTest.Data.ViewModels;

public class EmployeeViewModel : PersonViewModel
{
    public decimal? Salary { get; set; }
    public string? BankAccount { get; set; }

    public DateOnly? DateJoined { get; set; }
    public DateOnly? DateLeft { get; set; }

    public int? ManagerId { get; set; }
    public ShortManagerViewModel? Manager { get; set; }
}

File: UKParliament.CodeTest.Web/ClientApp/src/app/models/employee-view-model.ts
import { Address } from "./address";
import { BaseViewModel } from "./base-view-model";
import { ShortManagerViewModel } from "./short-manager-view-model";

export interface EmployeeViewModel extends BaseViewModel {
  firstName: string;
  lastName: string;
  employeeType: string;
  doB: Date,
  payBand: string,
  department: string,
  salary: number,
  bankAccount: string,

  dateJoined: Date,
  dateLeft: Date,

  manager: ShortManagerViewModel,

  address: Address,

  inactive: boolean,
  hasManager: boolean,
  isManager: boolean,
  hasIrregularities: boolean
}
